frame/two-phase-election-provider/src/benchmarking.rs

```diff
@@ -0,0 +1,358 @@
+		let caller = frame_benchmarking::whitelisted_caller();
+		T::Currency::make_free_balance_be(&caller,  T::Currency::minimum_balance() * 10u32.into());
+
+	}: _(RawOrigin::Signed(caller), solution, c)
+	verify {
+		assert!(<TwoPhase<T>>::signed_submissions().len() as u32 == c + 1);
+	}
+
+	submit_unsigned {
+		// number of votes in snapshot.
+		let v in 2000 .. 3000;
+		// number of targets in snapshot.
+		let t in 500 .. 800;
+		// number of assignments, i.e. compact.len(). This means the active nominators, thus must be
+		// a subset of `v` component.
+		let a in 500 .. 1500;
+		// number of desired targets. Must be a subset of `t` component.
+		let d in 200 .. 400;
+
+		let witness = SolutionSize { voters: v, targets: t };
+		let raw_solution = solution_with_size::<T>(witness, a, d);
+
+		assert!(<TwoPhase<T>>::queued_solution().is_none());
+		<CurrentPhase<T>>::put(Phase::Unsigned((true, 1u32.into())));
+	}: _(RawOrigin::None, raw_solution, witness)
+	verify {
+		assert!(<TwoPhase<T>>::queued_solution().is_some());
+	}
+
+	// This is checking a valid solution. The worse case is indeed a valid solution.
+	feasibility_check {
+		// number of votes in snapshot.
+		let v in 2000 .. 3000;
+		// number of targets in snapshot.
+		let t in 500 .. 800;
+		// number of assignments, i.e. compact.len(). This means the active nominators, thus must be
+		// a subset of `v` component.
+		let a in 500 .. 1500;
+		// number of desired targets. Must be a subset of `t` component.
+		let d in 200 .. 400;
+
+		let size = SolutionSize { voters: v, targets: t };
+		let raw_solution = solution_with_size::<T>(size, a, d);
+
+		assert_eq!(raw_solution.compact.voter_count() as u32, a);
+		assert_eq!(raw_solution.compact.unique_targets().len() as u32, d);
+	}: {
+		assert_ok!(<TwoPhase<T>>::feasibility_check(raw_solution, ElectionCompute::Unsigned));
+	}
+}
+
+#[cfg(test)]
+mod test {
+	use super::*;
+	use crate::mock::*;
+
+	#[test]
+	fn test_benchmarks() {
+		ExtBuilder::default().build_and_execute(|| {
+			assert_ok!(test_benchmark_feasibility_check::<Runtime>());
+		});
+
+		ExtBuilder::default().build_and_execute(|| {
+			assert_ok!(test_benchmark_submit_unsigned::<Runtime>());
+		});
+
+		ExtBuilder::default().build_and_execute(|| {
+			assert_ok!(test_benchmark_submit::<Runtime>());
+		});
+
+		ExtBuilder::default().build_and_execute(|| {
+			assert_ok!(test_benchmark_on_initialize_open_signed::<Runtime>());
+		});
+
+		ExtBuilder::default().build_and_execute(|| {
+			assert_ok!(test_benchmark_on_initialize_open_unsigned::<Runtime>());
+		});
+
+		ExtBuilder::default().build_and_execute(|| {
+			assert_ok!(test_benchmark_on_initialize_nothing::<Runtime>());
+		});
+
+		ExtBuilder::default().build_and_execute(|| {
+			assert_ok!(test_benchmark_finalize_signed_phase_accept_solution::<
+				Runtime,
+			>());
+		});
+
+		ExtBuilder::default().build_and_execute(|| {
+			assert_ok!(test_benchmark_finalize_signed_phase_reject_solution::<
+				Runtime,
+			>());
+		});
```

Comment:
I smell autoformatter here 👿

Reply:
I must admit that in PRs that I know are mostly touching new code (i.e. a new pallet like now) I do use rustfmt with a sensible config that is mostly aligned with our standards. In this case though, I see that this line would have been below the 100 char line width limit, so I am not sure what is going on.

In the future, I would be happy to _not_ do this practice if we establish it within Frame or whatever. Or the opposite 😈